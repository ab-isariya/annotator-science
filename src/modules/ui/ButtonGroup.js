/**
 * Buttons
 * figma: https://www.figma.com/file/d2iT8aM5wR0AN6ZHukOu5C/TAT-Style-Sheet-and-Master-Components?node-id=1986%3A17
 */
import React from 'react';
import PropTypes from 'prop-types';

const ButtonGroup = ({className, children}) => {
  return (
    <div
      className={`flex flex-row items-stretch overflow-hidden border-grey-500 border rounded-md h-full ${className}`}>
      {children}
    </div>
  );
};

ButtonGroup.propTypes = {
  //Group classname
  className: PropTypes.string,
  //Children nodes rendered inside
  children: PropTypes.node
};

export default ButtonGroup;
