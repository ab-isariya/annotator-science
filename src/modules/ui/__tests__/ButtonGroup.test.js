import ButtonGroup from '../ButtonGroup';
import renderer from 'react-test-renderer';

describe('<ButtonGroup/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer.create(<ButtonGroup />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with children', () => {
    const comp = renderer.create(<ButtonGroup>Hello</ButtonGroup>).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.children).not.toEqual(null);
  });

  it('Renders with a custom classname', () => {
    const comp = renderer
      .create(<ButtonGroup className="testClass" />)
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });
});
