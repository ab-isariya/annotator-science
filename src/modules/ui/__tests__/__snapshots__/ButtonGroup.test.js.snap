// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup/> Renders with a custom classname 1`] = `
<div
  className="flex flex-row items-stretch overflow-hidden border-grey-500 border rounded-md h-full testClass"
/>
`;

exports[`<ButtonGroup/> Renders with children 1`] = `
<div
  className="flex flex-row items-stretch overflow-hidden border-grey-500 border rounded-md h-full undefined"
>
  Hello
</div>
`;

exports[`<ButtonGroup/> Renders without crashing 1`] = `
<div
  className="flex flex-row items-stretch overflow-hidden border-grey-500 border rounded-md h-full undefined"
/>
`;
